reana_cluster/backends/kubernetes/k8s.py

```diff
@@ -0,0 +1,510 @@
+    # TODO: Docstring
+
+    __cluster_type = 'kubernetes'
+
+    _conf = {
+        'templates_folder': pkg_resources.resource_filename(
+            __name__, '/templates'),
+        'min_version': 'v1.6.4',
+        'max_version': 'v1.6.4',
+    }
+
+    def __init__(self,
+                 cluster_spec,
+                 cluster_conf=None,
+                 kubeconfig=None,
+                 context=None):
+
+        # Load Kubernetes cluster configuration. Defaults to '~/.kube/config'
+        k8s_config.load_kube_config(kubeconfig, context)
+
+        # Instantiate clients for various Kubernetes REST APIs
+        self.corev1api = k8s_client.CoreV1Api()
+        self.versionapi = k8s_client.VersionApi()
+        self.extbetav1api = k8s_client.ExtensionsV1beta1Api()
+
+        self.cluster_spec = cluster_spec
+        self.cluster_conf = self.generate_configuration(cluster_spec)
+
+    @property
+    def cluster_type(self):
+        return self.__cluster_type
+
+    @property
+    def cluster_type(self):
+        return self.cluster_conf
+
+    @property
+    def current_config(self):
+        # TODO: Check what conf holds when `load_kube_config()` hasn't been run
+        if configuration is None:
+            # TODO: Raise somekind of "NotInitialized" error is
+            # load_kube_config() has not been run.
+            pass
+        # TODO: Probably better to return only relevant data not the whole conf
+        return configuration
+
+    @classmethod
+    def generate_configuration(cls, cluster_spec):
+        """Generate Kubernetes backend configuration files to init REANA cluster.
+
+        :param cluster_spec: Dictionary representing complete REANA
+            cluster spec file.
+        """
+
+        # Setup an Environment for Jinja
+        env = Environment(
+            loader=FileSystemLoader(
+                cls._conf['templates_folder']),
+            keep_trailing_newline=False
+        )
+
+        # Define where are backend conf params needed when rendering templates.
+        be_conf_params_fp = cls._conf['templates_folder'] + '/config.yaml'
+
+        try:
+            with open(be_conf_params_fp) as f:
+
+                # Load backend conf params
+                backend_conf_parameters = yaml.load(f.read())
+
+                # Load template combining all other templates from
+                # templates folder
+                # TODO: Better to combine templates or populated templates in
+                # Python code for improved extensibility?
+                # Just drop a .yaml template and add necessary to config.yaml
+                # without changing anything?
+                # template = env.get_template('deployments/complete.yaml')
+                template = env.get_template('complete.yaml')
+
+                components = cluster_spec['components']
+                server_img = components['reana-server']['image']
+                jc_img = components['reana-job-controller']['image']
```

Comment:
For these abbreviations I would go for the first letter of each word:
`reana-server` -> `rs`
`reana-job-controller` -> `rjc`
....
`reana-workflow-engine-yadage` -> `rwey`
`reana-workflow-engine-cwl` -> `rwecwl` or `rwec`

what do you think? Just to have some naming consistency across the repos (i.e. making references in commit messages).

Reply:
Sure sounds good.

Based on previous experience, I did a quick check with Google and Twitter and nothing really controversial seems to be associated to them :P
```
rmb = https://en.wikipedia.org/wiki/Renminbi
rjc = https://twitter.com/rjc
```

> ```
> reana-workflow-engine-yadage -> rwey
> reana-workflow-engine-cwl -> rwecwl or rwec
> ```
For these I might even go with `rwe_y` and `rwe_cwl` just to highlight the *REANA* -part.